packages/opencensus-core/src/trace/instrumentation/types.ts

```diff
@@ -24,31 +24,45 @@ export interface Plugin {
    * @param moduleExports nodejs module exports from the module to patch
    * @param tracer a tracer instance
    * @param version version of the current instaled module to patch
+   * @param options plugin options
    * @param basedir module absolute path
    */
   enable(
       // tslint:disable-next-line:no-any
       moduleExports: any, tracer: Tracer, version: string,
+      options: PluginConfig,
       // tslint:disable-next-line:no-any
       basedir?: string): any;
   /** Method to disable the instrumentation  */
   disable(): void;
 }
 
+export type PluginConfig = {
+  // tslint:disable-next-line:no-any
+  [key: string]: any;
+};
+
+export type NamedPluginConfig = {
```

Comment:
Should we have:
```
module: string;
config: PluginConfig;
```
That way there's no confusion over what is the module name and what the config is. There's no room for name collision.